### ISSUE TITLE
Switch Element: aria-labelledby not resolving

### ISSUE THREAD

**Issue by ianmcburnie** [makeup/makeup-ui#1]
It seems that the element id referenced via `aria-labelledby` cannot be resolved. Possibly due to shadow DOM boundary restrictions...? May have to use `aria-label` instead, or refactor to incorporate the labelling element inside of the web component itself. Requires more investigation.

https://makeup.github.io/makeup-ui/makeup-switch-element/index.html

<img width="1419" alt="Screen Shot 2020-04-04 at 11 06 28 AM" src="https://user-images.githubusercontent.com/38065/78458211-6d40ec80-7664-11ea-93ef-755bb7c8ff36.png">

<img width="1419" alt="Screen Shot 2020-04-04 at 11 07 18 AM" src="https://user-images.githubusercontent.com/38065/78458214-76ca5480-7664-11ea-939d-db6441270240.png">

Accessibility inspector:

<img width="488" alt="Screen Shot 2020-04-04 at 11 11 18 AM" src="https://user-images.githubusercontent.com/38065/78458297-18ea3c80-7665-11ea-8320-cc403b17136f.png">

When a labelling element is added inside of Shadow DOM:

<img width="490" alt="Screen Shot 2020-04-04 at 11 10 52 AM" src="https://user-images.githubusercontent.com/38065/78458296-138cf200-7665-11ea-9333-d823add0a05b.png">

**Comment by ianmcburnie** [comment]
Seems like this is indeed a limitation/restriction put in place by the browser (thank you Erik Kroes and Paul Grenier for assisting me).

https://developer.paciellogroup.com/blog/2012/07/notes-on-web-components-aria/#comment-726

As a workaround I am using the underlying class to grab a reference to the labelling element, scraping its inner text, and setting aria-label. This routine will happen every time the aria-labelledby value is updated.

p.s. good article on a11y & web components by Erik Kroes:

https://www.erikkroes.nl/blog/accessibility/the-guide-to-accessible-web-components-draft/index.html

This article by Léonie Watson mentions the shadow boundary problem directly:

https://www.24a11y.com/2019/web-components-and-the-aom/
